synapse/rest/media/v1/thumbnail_resource.py

```diff
@@ -379,8 +392,40 @@ async def _select_and_respond_with_thumbnail(
                 file_info.thumbnail.length,
             )
         else:
+            # This might be because:
+            # 1. We can't create thumbnails for the given media (corrupted or
+            #    unsupported file type)
+            # 2. The thumbnailing process never ran or errored out initially
+            #    when the media was first uploaded (these bugs should be
+            #    reported and fixed).
+            # 3. `dynamic_thumbnails` is disabled (see `homeserver.yaml`) and we
+            #    can't try to generate one. If this was enabled, Synapse would
+            #    go down a different code path.
             logger.info("Failed to find any generated thumbnails")
-            respond_404(request)
+
+            # We could get away with putting this error text directly in the
+            # error below because it never hits this code path otherwise when
+            # `dynamic_thumbnails` is disabled. But it would be good not to
+            # always assume this will be the case in the future so we provide
+            # accurate information to the user.
+            dynamic_thumbnails_disabled_warning_text = ""
+            if not self.dynamic_thumbnails:
+                dynamic_thumbnails_disabled_warning_text = " We also cannot try to generate a new thumbnail because `dynamic_thumbnails` is disabled (see `homeserver.yaml`)."
+
+            respond_with_json(
+                request,
+                400,
```

Comment:
The spec says no right now, https://github.com/matrix-org/matrix-spec/issues/1122

It does make sense to me in some cases:

> `400` makes a lot of sense for all of the scenarios where the media repository can't provide a thumbnail for whatever reason when the media exists.
> 
> But `404` probably makes more sense when `mediaId` isn't valid and doesn't exist.

In the cases I ran into, `400` makes more sense since the media did exist, Synapse just didn't thumbnail it.